hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java

```diff
@@ -351,38 +363,27 @@ private void handleOutputStreamWrite(int currIdx, long len,
       try {
         // Since it's a fullcell, let's write all content from buffer.
         writeToOutputStream(current, len, bytesToWrite.array(),
-            bytesToWrite.array().length, 0, current.getWrittenDataLength(),
-            isParity);
+            bytesToWrite.array().length, 0, isParity);
       } catch (Exception e) {
         markStreamAsFailed(e);
       }
     }
   }
 
   private int writeToOutputStream(BlockOutputStreamEntry current, long len,
-      byte[] b, int writeLen, int off, long currentPos, boolean isParity)
+      byte[] b, int writeLen, int off, boolean isParity)
       throws IOException {
     try {
       current.write(b, off, writeLen);
       if (!isParity) {
         offset += writeLen;
       }
     } catch (IOException ioe) {
-      // for the current iteration, totalDataWritten - currentPos gives the
-      // amount of data already written to the buffer
-
-      // In the retryPath, the total data to be written will always be equal
-      // to or less than the max length of the buffer allocated.
-      // The len specified here is the combined sum of the data length of
-      // the buffers
-      Preconditions.checkState(len <= config.getStreamBufferMaxSize());
-      int dataWritten = (int) (current.getWrittenDataLength() - currentPos);
-      writeLen = dataWritten;
-
       if (!isParity) {
```

Comment:
if we remove the changes to writeLen, during the exception handling, then we can just increase the offset before the write in the try block.
Though I am not sure why this change is happening, can you share some reasoning? The original idea here is that there might be a partial write, and if that happened, we wanted to decrease the reported written length, with the change we do not do so anymore. Shouldn't we care about it? You might be right, and probably I just don't see it :)

Reply:
Sure, I can move that part to above try block. Since the error checking and retrying within the stream is not implemented yet, all the length adjustments happen at the time of handleStripeFailure. So, we will just increment the length here and if there is an exception, setIOException would be invoked inside handleException. Later before and after putBlock, we will check stream failure statuses.

Reply:
thank you, that sounds good.